test/helpers/create-https-test-server.ts

```diff
@@ -0,0 +1,72 @@
+import https = require('https');
+import net = require('net');
+import express = require('express');
+import pify = require('pify');
+import pem = require('pem');
+
+export type HttpsServerOptions = {
+	commonName?: string;
+	days?: number;
+};
+
+export interface ExtendedHttpsTestServer extends express.Express {
+	https: https.Server;
+	caKey: Buffer;
+	caCert: Buffer;
+	url: string;
+	port: number;
+	close: () => Promise<any>;
+}
+
+const createHttpsTestServer = async (options: HttpsServerOptions = {}): Promise<ExtendedHttpsTestServer> => {
+	const createCSR = pify(pem.createCSR);
+	const createCertificate = pify(pem.createCertificate);
+
+	const caCSRResult = await createCSR({commonName: 'authority'});
+	const caResult = await createCertificate({
+		csr: caCSRResult.csr,
+		clientKey: caCSRResult.clientKey,
+		selfSigned: true
+	});
+	const caKey = caResult.clientKey;
+	const caCert = caResult.certificate;
+
+	const serverCSRResult = await createCSR({commonName: options.commonName ?? 'localhost'});
+	const serverResult = await createCertificate({
+		csr: serverCSRResult.csr,
+		clientKey: serverCSRResult.clientKey,
+		serviceKey: caKey,
+		serviceCertificate: caCert,
+		days: options.days ?? 365
+	});
+	const serverKey = serverResult.clientKey;
+	const serverCert = serverResult.certificate;
+
+	const server = express() as ExtendedHttpsTestServer;
+	server.https = https.createServer(
```

Comment:
Do we need to set the `https` property?

Reply:
Ah, yes, you're right.

Reply:
Maybe let's rename `server` to `app` and `server.https` to `app.server`? Does Express allow us to access the underlying server some other way?

Reply:
HTTPS is not 100% needed, but it could be useful to have it there.
I don't think there's a way to access the server from Express because the HTTPS Server calls Express as a function, so the connection is like `Server => Express` and not `Express => Server`.

The rename is ok for me, I had just copied the lukechilds notation.

Reply:
Since this conversation is marked as resolved I assume the rename is not needed, am I correct?